### PR TITLE
feat: add option to allow version locking of js integrations

### PR DESCRIPTION
**Pull request by bardisg**
## PR Description

Allow the user to lock the version of the integrations to the same version of the SDK package.
Deploy the js-integrations in the versioned folders also (alongside v1.1 and latest)
Fix linting issues so the team stops using --no-verify

## Notion ticket

[Ticket link](https://www.notion.so/rudderstacks/Allow-the-SDK-consumers-to-lock-the-version-of-integrations-398aa92ffb494f99b33a3823c8be446d)

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/813)

### CODE REVIEW

File: src/utils/cdnPaths.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  if (customIntegrationsCDNPath) {
+    integrationsCDNPath = removeTrailingSlashes(customIntegrationsCDNPath);
+
+    // TODO: add proper url validation
+    if (!integrationsCDNPath) {
+      handleError({
+        message: '[Analytics] load:: CDN base URL is not valid',
+      });
+      throw Error('Failed to load Rudder SDK from provided URL');
+    }
```
Comment (saikumarrs) [inline src/utils/cdnPaths.js:23]:
```suggestion
      const errMsg = 'CDN base URL for integrations is not valid';
      handleError({
        message: errMsg,
      });
      throw Error(`Failed to load RudderStack SDK: "${errMsg}"`);
    }
```

File: src/utils/RudderApp.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Application class
 class RudderApp {
   constructor() {
-    this.build = '1.0.0';
+    this.build = '1.1.0';
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/RudderApp.js:4]: @saikumarrs should I remove it? is was there pointing in wrong arch version

File: src/utils/cdnPaths.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { handleError } from './errorHandler';
+import { CDN_ARCH_VERSION_DIR, CDN_INT_DIR, DEST_SDK_BASE_URL } from './constants';
+
+const getIntegrationsCDNPath = (
+  currentVersion,
+  lockIntegrationsVersion,
+  customIntegrationsCDNPath,
+) => {
+  let integrationsCDNPath = '';
+  const { sdkURL } = getSDKUrlInfo();
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/cdnPaths.js:11]: done

File: src/utils/cdnPaths.js (hunk 1 above)
Comment (bardisg) [reply src/utils/cdnPaths.js:23]: done

File: src/utils/RudderApp.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bardisg) [reply src/utils/RudderApp.js]: removed it

**Review by MoumitaM — APPROVED** [review]

File: __tests__/utils/cdnPaths.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  });
+
+  it('should throw error if invalid custom url is provided', () => {
+    const integrationsCDNPath = () => getIntegrationsCDNPath(dummyVersion, false, '/');
+    expect(integrationsCDNPath).toThrow('Failed to load Rudder SDK from provided URL');
+  });
+
+  it('should return script src path if script src exists and integrations version is not locked', () => {
+    const integrationsCDNPath = getIntegrationsCDNPath(dummyVersion, false);
+    expect(integrationsCDNPath).toBe('https://www.dummy.url/fromScript/v1.1/js-integrations');
```
[the remark this answers is not in the document]
Comment (bardisg) [reply __tests__/utils/cdnPaths.test.js:42]: dummyScriptURL is not the same as the expected. it is te full path to the analytics javascript file. The expected it the basepath /js-integrations

**Pull request merged**